presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilter.java

```diff
@@ -112,7 +123,17 @@ private synchronized void addPartition(TupleDomain<String> tupleDomain)
         return builder.build();
     }
 
-    public static Optional<LocalDynamicFilter> create(JoinNode planNode, TypeProvider types, int partitionCount)
+    private Map<String, Domain> convertTupleDomain(TupleDomain<String> result)
+    {
+        if (result.isNone()) {
+            // One of the join build symbols has no non-null values, therefore no filters can match predicate
+            return buildChannels.keySet().stream()
+                    .collect(toImmutableMap(identity(), filterId -> Domain.none(filterBuildTypes.get(filterId))));
+        }
+        return result.getDomains().get();
+    }
+
+    public static Optional<LocalDynamicFilter> create(JoinNode planNode, List<Type> buildSourceTypes, int partitionCount)
```

Comment:
I don't think this needs to return `Optional` anymore. You could assert that `join.getDynamicFilters()` is not empty

Reply:
Right, made this change now